src/common/iconbndl.cpp

```diff
@@ -192,6 +206,75 @@ void wxIconBundle::AddIcon(wxInputStream& stream, wxBitmapType type)
 
 #endif // wxUSE_STREAMS && wxUSE_IMAGE
 
+
+#ifdef __WINDOWS__
+
+namespace {
+
+// struct declarations taken from https://msdn.microsoft.com/en-us/library/ms997538.aspx
```

Comment:
I actually discovered those struct declarations when writing this code but as you said, they are not exactly the same, even though the only real difference is the lack of the entries array in the `ICONDIR` (and possibly packing). 
While I am not a fan of duplicated code, I decided that factoring them out was not worth it. Truth to be told, I do not know how I would use the `ICONDIR` both for the file and resource code.